src/ripple/net/impl/RegisterSSLCerts.cpp

```diff
@@ -27,32 +34,66 @@ namespace ripple {
 void
 registerSSLCerts(boost::asio::ssl::context& ctx, boost::system::error_code& ec)
 {
-#if BEAST_WINDOWS
-    HCERTSTORE hStore = CertOpenSystemStore(0, "ROOT");
-    if (hStore == NULL)
+#if BOOST_OS_WINDOWS
+    auto certStoreDelete = [](void* h) {
+        if (h != nullptr)
+            CertCloseStore(h, 0);
+    };
+    std::unique_ptr<void, decltype(certStoreDelete)> hStore{
+        CertOpenSystemStore(0, "ROOT"), certStoreDelete};
+
+    if (!hStore)
+    {
+        ec = boost::system::error_code(
+            GetLastError(), boost::system::system_category());
+        return;
+    }
+
+    ERR_clear_error();
+
+    std::unique_ptr<X509_STORE, decltype(X509_STORE_free)*> store{
+        X509_STORE_new(), X509_STORE_free};
+
+    if (!store)
     {
+        ec = boost::system::error_code(
+            static_cast<int>(::ERR_get_error()),
+            boost::asio::error::get_ssl_category());
         return;
     }
 
-    X509_STORE* store = X509_STORE_new();
     PCCERT_CONTEXT pContext = NULL;
-    while ((pContext = CertEnumCertificatesInStore(hStore, pContext)) != NULL)
+    while ((pContext = CertEnumCertificatesInStore(hStore.get(), pContext)) !=
+           NULL)
     {
-        X509* x509 = d2i_X509(
-            NULL,
-            (const unsigned char**)&pContext->pbCertEncoded,
-            pContext->cbCertEncoded);
-        if (x509 != NULL)
+        const unsigned char* pbCertEncoded = pContext->pbCertEncoded;
+        std::unique_ptr<X509, decltype(X509_free)*> x509{
+            d2i_X509(NULL, &pbCertEncoded, pContext->cbCertEncoded), X509_free};
+        if (!x509)
+        {
+            ec = boost::system::error_code(
+                static_cast<int>(::ERR_get_error()),
+                boost::asio::error::get_ssl_category());
+            return;
```

Comment:
This is ok as is (returning with `ec` and throwing if _anything_ goes wrong), but I'm wondering if we could alternatively be ok calling `continue` here (and on line 85) and just having the caller log `ec`.